godtools/App/Features/Menu/Presentation/DeleteAccountProgress/DeleteAccountProgressViewModel.swift

```diff
@@ -0,0 +1,61 @@
+//
+//  DeleteAccountProgressViewModel.swift
+//  godtools
+//
+//  Created by Levi Eggert on 5/23/23.
+//  Copyright © 2023 Cru. All rights reserved.
+//
+
+import Foundation
+import Combine
+
+class DeleteAccountProgressViewModel: ObservableObject {
+    
+    private let deleteAccountUseCase: DeleteAccountUseCase
+    private let localizationServices: LocalizationServices
+    
+    private var cancellables: Set<AnyCancellable> = Set()
+    
+    private weak var flowDelegate: FlowDelegate?
+    
+    @Published var title: String
+    
+    init(flowDelegate: FlowDelegate, deleteAccountUseCase: DeleteAccountUseCase, localizationServices: LocalizationServices) {
+        
+        self.flowDelegate = flowDelegate
+        self.deleteAccountUseCase = deleteAccountUseCase
+        self.localizationServices = localizationServices
+        
+        title = localizationServices.stringForMainBundle(key: "deleteAccountProgress.title")
+             
+        deleteAccountUseCase.deleteAccountPublisher()
+            .receiveOnMain()
+            .sink { [weak self] subscribersCompletion in
+                
+                let deleteAccountError: Error?
+                
+                switch subscribersCompletion {
+                    
+                case .finished:
+                    deleteAccountError = nil
+                    
+                case .failure(let error):
+                    deleteAccountError = error
+                }
+                
+                DispatchQueue.main.async {
+                    
+                    if let deleteAccountError = deleteAccountError {
+                        self?.flowDelegate?.navigate(step: .didFinishAccountDeletionWithErrorFromDeleteAccountProgress(error: deleteAccountError))
+                    }
+                    else {
+                        self?.flowDelegate?.navigate(step: .didFinishAccountDeletionWithSuccessFromDeleteAccountProgress)
+                    }
+                }
```

Comment:
Shouldn't we already be on main here?

Reply:
Hey @rachaelblue good catch!  For some reason I did have a thread crash here earlier when implementing, but maybe I didn't have the ```receiveOnMain()``` in place.  I can't exactly remember now, but after running the app a few times it is calling on the mainThread so will remove the DispatchQueue.main.async.

Reply:
Hey @rachaelblue right! I don't think the delete account option should show if you're not signed in.  And I just realized too that in order for a user to actually delete their account they will need a valid mobile-content-api access token.  Going to create a separate ticket to hide the Delete Account menu option.